visualizations/frontend/filters/v2/TDatePicker.vue

```diff
@@ -1,322 +1,328 @@
+      default: " ~ ",
+    },
+    defaultStartDate: {
+      type: String,
+      default: "",
+    },
+    defaultEndDate: {
+      type: String,
+      default: "",
+    },
+    defaultDatePreset: {
+      type: String,
+      default: "",
+    },
+  },
+  computed: {
+    mode() {
+      if (
+        this.$attrs["t-filter:start_date"] &&
+        this.$attrs["t-filter:end_date"]
+      ) {
+        return "range";
+      }
+      return "single";
+    },
+  },
+  data: () => ({
+    date: [],
+    presets: {
+      range: [
+        { key: "last7days", label: "Last 7 Days" },
+        { key: "lastmonth", label: "Last Month" },
+        { key: "last30days", label: "Last 30 Days" },
+        { key: "last90days", label: "Last 90 Days" },
+        { key: "mtd", label: "Month to Date" },
+        { key: "ytd", label: "Year to Date" },
+        { key: "custom", label: "Custom Range" },
+      ],
+      single: [
+        { key: "today", label: "Today" },
+        { key: "yesterday", label: "Yesterday" },
+        { key: "7DaysAgo", label: "7 Days Ago" },
+        { key: "30DaysAgo", label: "30 Days Ago" },
+        { key: "90DaysAgo", label: "90 Days Ago" },
+        { key: "yearAgo", label: "A Year Ago" },
+        { key: "custom", label: "Custom Date" },
+      ],
     },
-    computed: {
-        mode() {
-            if (this.$attrs['t-filter:start_date'] && this.$attrs['t-filter:end_date']) {
-                return "range";
-            }
-            return "single";
+    selectedPreset: {},
+    is_filter: true,
+  }),
+  methods: {
+    onVisualizationInit() {
+      this.assignDatesFromFilters();
+      this.handleDefaults();
+    },
+    setCustomPreset() {
+      this.selectedPreset = this.presets[this.mode].filter(
+        (p) => p.key === "custom"
+      )[0];
+    },
+    handleConfirm(date) {
+      // Url should always store date in a readable format to avoid failure on component load.
+      if (this.mode === "range") {
+        // Returns array
+        const startDate = date[0];
+        const endDate = date[1];
+        this.setFilterValue(
+          "start_date",
+          this.formatDate(startDate, "YYYY-MM-DD")
+        );
+        this.setFilterValue("end_date", this.formatDate(endDate, "YYYY-MM-DD"));
+      } else {
+        // Returns string
+        this.setFilterValue("start_date", this.formatDate(date, "YYYY-MM-DD"));
+      }
+      this.setFilterValue("date_preset", this.selectedPreset.key);
+    },
+    assignDatesFromFilters() {
+      const preset = this.getFilterValue("date_preset");
+      if (preset) {
+        const presetObject = this.presets[this.mode].filter(
+          (p) => p.key === preset
+        )[0];
+        this.selectedPreset = presetObject;
+
+        if (presetObject.key !== "custom") {
+          // If relative date, use the date assigned by preset.
+          this.handlePreset(presetObject);
+          this.handleConfirm(this.date);
+          return;
+        }
+      }
```

Comment:
Change 1:

Preset was being set after the dates before, now it will check if the selected preset was not `custom` then it will calculate the relative date according to selected preset instead of keeping dates absolute.